metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -143,14 +140,14 @@ public void execute(Tuple tuple) {
       boolean ackTuple = true;
       if(sensorParserConfig != null) {
         List<FieldValidator> fieldValidations = getConfigurations().getFieldValidations();
-        List<JSONObject> messages = parser.parse(originalMessage);
-        for (JSONObject message : messages) {
-          if (parser.validate(message)) {
+        Optional<List<JSONObject>> messages = parser.parseOptional(originalMessage);
+        for (JSONObject message : messages.orElse(Collections.emptyList())) {
+          if (parser.validate(message) && filter != null && filter.emitTuple(message)) {
             if(!isGloballyValid(message, fieldValidations)) {
               message.put(Constants.SENSOR_TYPE, getSensorType()+ ".invalid");
               collector.emit(Constants.INVALID_STREAM, new Values(message));
             }
-            else if (filter != null && filter.emitTuple(message)) {
+            else {
               ackTuple = !isBulk;
```

Comment:
The tuple will be acked.  Note that ackTuple is initialized to true and is only set to false if we are in a situation where we are doing a bulk write, in which case the tuple will be acked in the bulk writer upon the writing of a batch of entries.  So, when this loop is ended, it will ack on line 164.